content/blog/using-the-hpegl-hpe-greenlake-and-morpheus-terraform-providers-in-concert-to-create-provision-and-manage-hpe-pce-private-cloud-enterprise-vmaas-instances.md

```diff
@@ -0,0 +1,1494 @@
+      + result = (known after apply)
+    }
+
+Plan: 6 to add, 0 to change, 0 to destroy.
+morpheus_node_type.tf_example_node: Creating...
+morpheus_instance_type.tf_example_instance_type: Creating...
+morpheus_group.tf_example_group: Creating...
+random_integer.random: Creating...
+random_integer.random: Creation complete after 0s [id=38926]
+morpheus_instance_type.tf_example_instance_type: Creation complete after 2s [id=208]
+morpheus_node_type.tf_example_node: Creation complete after 2s [id=1969]
+morpheus_instance_layout.tf_example_instance_layout: Creating...
+morpheus_group.tf_example_group: Creation complete after 2s [id=84]
+morpheus_instance_layout.tf_example_instance_layout: Creation complete after 2s [id=1410]
+hpegl_vmaas_instance.sample_vm[0]: Creating...
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [10s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [20s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [30s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [40s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [50s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [1m0s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [1m10s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [1m20s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Still creating... [1m30s elapsed]
+hpegl_vmaas_instance.sample_vm[0]: Creation complete after 1m34s [id=52689]
+
+Apply complete! Resources: 6 added, 0 changed, 0 destroyed.
+```
+
+## One Location/Two VMaaS Clouds/Two VMaaS Instances example
+
+![](/img/hpegl-and-morpheus-terraform-blog-post-frame-2-1-.jpg)
+
+The example HCL used here is based on the HCL for the first example.  However we have created two separate *Modules* based on that HCL: 
+
+* *morpheus_artefacts* which does the following 
+
+  * Takes as input: 
+
+    * A VMaaS Cloud Name 
+    * A Group name 
+    * An image name 
+  * Creates: 
+
+    * A Group 
+    * An Instance Type 
+    * An Instance Layout 
+    * A Node Type 
+  * Outputs: 
+
+    * Instance Type details 
+    * Instance Layout details 
+    * Node Type details 
+    * Group details 
+    * Cloud ID 
+* *vmaas_instance* which does the following 
+
+  * Takes inputs from an instance of the “morpheus_artefacts” module 
+  * Retrieves details for the following VMaaS resources: 
+
+    * VMaaS Datastore 
+    * VMaaS Network 
+    * VMaaS Resource Pool 
+    * VMaaS Plan 
+    * VMaaS Environment 
+    * VMaaS Cloud Folder 
+  * Creates a VMaaS instance 
+
+These modules can be combined in different ways.  In our specific case we: 
+
+* Make two calls to *morpheus_artefacts* 
+* Make two calls to *vmaas_instance* each using outputs from one of the above calls to *morpheus_artefacts* 
+
+In this way we create two VMaaS instances each in a separate VMaaS Cloud based on the one underlying on-premise Morpheus instance.
+
+### HCL
+
+< include link to repo contents here >
```

Comment:
Needs link.

Reply:
I'll add that once the [PR](https://github.com/hpe-dev-incubator/GLP-API-Tooling/pull/4) has landed

Reply:
Done.